python/runtime/cudaq/operators/py_boson_op.cpp

```diff
@@ -0,0 +1,498 @@
+  .def("__sub__", [](const boson_op &self, int other) { return self - other; }, py::is_operator())
+  .def("__truediv__", [](const boson_op &self, const scalar_operator &other) { return self / other; }, py::is_operator())
+  .def("__mul__", [](const boson_op &self, const scalar_operator &other) { return self * other; }, py::is_operator())
+  .def("__add__", [](const boson_op &self, const scalar_operator &other) { return self + other; }, py::is_operator())
+  .def("__sub__", [](const boson_op &self, const scalar_operator &other) { return self - other; }, py::is_operator())
+  .def("__mul__", [](const boson_op &self, const boson_op_term &other) { return self * other; }, py::is_operator())
+  .def("__add__", [](const boson_op &self, const boson_op_term &other) { return self + other; }, py::is_operator())
+  .def("__sub__", [](const boson_op &self, const boson_op_term &other) { return self - other; }, py::is_operator())
+  .def("__mul__", [](const boson_op &self, const boson_op &other) { return self * other; }, py::is_operator())
+  .def("__add__", [](const boson_op &self, const boson_op &other) { return self + other; }, py::is_operator())
+  .def("__sub__", [](const boson_op &self, const boson_op &other) { return self - other; }, py::is_operator())
+  .def("__mul__", [](const boson_op &self, const matrix_op_term &other) { return self * other; }, py::is_operator())
+  .def("__add__", [](const boson_op &self, const matrix_op_term &other) { return self + other; }, py::is_operator())
+  .def("__sub__", [](const boson_op &self, const matrix_op_term &other) { return self - other; }, py::is_operator())
+  .def("__mul__", [](const boson_op &self, const matrix_op &other) { return self * other; }, py::is_operator())
+  .def("__add__", [](const boson_op &self, const matrix_op &other) { return self + other; }, py::is_operator())
+  .def("__sub__", [](const boson_op &self, const matrix_op &other) { return self - other; }, py::is_operator())
+
+  // left-hand arithmetics
+
+  .def("__rmul__", [](const boson_op &other, int self) { return self * other; }, py::is_operator())
+  .def("__radd__", [](const boson_op &other, int self) { return self + other; }, py::is_operator())
+  .def("__rsub__", [](const boson_op &other, int self) { return self - other; }, py::is_operator())
+  .def("__rmul__", [](const boson_op &other, double self) { return self * other; }, py::is_operator())
+  .def("__radd__", [](const boson_op &other, double self) { return self + other; }, py::is_operator())
+  .def("__rsub__", [](const boson_op &other, double self) { return self - other; }, py::is_operator())
+  .def("__rmul__", [](const boson_op &other, std::complex<double> self) { return self * other; }, py::is_operator())
+  .def("__radd__", [](const boson_op &other, std::complex<double> self) { return self + other; }, py::is_operator())
+  .def("__rsub__", [](const boson_op &other, std::complex<double> self) { return self - other; }, py::is_operator())
+
```

Comment:
It seems like we missed left-hand arithmetic overloads for `boson_op` with `scalar_operator`.